### PR TITLE
Update covid consensus wf

### PR DESCRIPTION
**Pull request by wm75**
This is the fixed and improved version that's used by the COG-UK and Estonian data analysis bot on .eu since Friday.

### CODE REVIEW

**Comment by wm75** [comment]
@mvdbeek That weird issue where the `tool_version` gets dropped and set to `null` stroke again when I updated this WF. I put the correct versions back in, but I don't understand why this is still an issue in the first place. I thought this is fixed since quite some time now?

**Comment by wm75** [comment]
I just noticed that the current version of the WF did not get a v0.1 tag at dockstore. Did I forget to do something to make this happen?

**Comment by mvdbeek** [comment]
No, the dockstore integration is sometimes missing the release tag.

> That weird issue where the `tool_version` gets dropped and set to `null` stroke again when I updated this WF.

You've changed the version in the editor, and when you download the workflow the tool_version is null ? I'll see if I can find out what's up with that.

Also the test itself failed with
```
galaxy.jobs.runners ERROR 2021-05-05 14:04:25,369 [pN:main,p:6063,tN:LocalRunner.work_thread-1] (2/) Job wrapper finish method failed
Traceback (most recent call last):
  File "/tmp/tmpuarb3tc3/galaxy-dev/lib/galaxy/jobs/runners/__init__.py", line 541, in _finish_or_resubmit_job
    job_wrapper.finish(tool_stdout, tool_stderr, exit_code, check_output_detected_state=check_output_detected_state, job_stdout=job_stdout, job_stderr=job_stderr)
  File "/tmp/tmpuarb3tc3/galaxy-dev/lib/galaxy/jobs/__init__.py", line 1689, in finish
    output_name, dataset, job, context, final_job_state, remote_metadata_directory
  File "/tmp/tmpuarb3tc3/galaxy-dev/lib/galaxy/jobs/__init__.py", line 1546, in _finish_dataset
    dataset.datatype.set_meta(dataset, overwrite=False)
  File "/tmp/tmpuarb3tc3/galaxy-dev/lib/galaxy/datatypes/binary.py", line 549, in set_meta
    pysam.index(dataset.file_name, index_file.file_name)
  File "/home/runner/.planemo/gx_venv_3.7/lib/python3.7/site-packages/pysam/utils.py", line 75, in __call__
    stderr))
pysam.utils.SamtoolsError: 'samtools returned with error 1: stdout=, stderr=samtools index: "/tmp/tmpuarb3tc3/files/000/dataset_2.dat" is in a format that cannot be usefully indexed\n'
```
Is the bam file coordinate sorted ?

**Comment by wm75** [comment]
> Is the bam file coordinate sorted ?

Saw the error, yes, and just checked. I can *samtools index* the file locally without an issue.

**Comment by wm75** [comment]
Could this have to do with my attempt at specifying the dataset types in the yml?

**Comment by mvdbeek** [comment]
> Could this have to do with my attempt at specifying the dataset types in the yml?

It could, if the file is not coordinate sorted. Let me re-run the workflow to be sure.

**Comment by mvdbeek** [comment]
```
In [1]: import pysam

In [2]: pysam.index('aligned_reads_for_coverage.bam')
Out[2]: ''
```
the file is fine as you said, and pysam doens't seem to have a problem either. Very strange.

**Comment by wm75** [comment]
> You've changed the version in the editor, and when you download the workflow the tool_version is null ? I'll see if I can find out what's up with that.

I changed the WF on .eu to prepare this PR and, at some point, when it became clear that the changes would be more than trivial, I saved it under a new name. When I finally downloaded the new version it had all tool_versions set to null.

**Comment by mvdbeek** [comment]
That is a good hint, thanks!

**Comment by wm75** [comment]
Should I just try without the bam datatype specified?

**Comment by mvdbeek** [comment]
Yeah, give it a try

**Comment by mvdbeek** [comment]
> I changed the WF on .eu to prepare this PR and, at some point, when it became clear that the changes would be more than trivial, I saved it under a new name. When I finally downloaded the new version it had all tool_versions set to null.

Ugh ... this seems to work on my machine for some trivial edits. It gets fun on .org ... the tool_version changes work, but the tool_id doesn't get updated and the handlers don't all have the latest version of `bedtools_genomecoveragebed`.

**Comment by mvdbeek** [comment]
I'm rerunning the last successful test in https://github.com/galaxyproject/iwc/actions/runs/799978503 ... let's see if that still works.

**Comment by wm75** [comment]
Still works. Debugging could be easier :(

File: workflows/sars-cov-2-variant-calling/sars-cov-2-consensus-from-variation/consensus-from-variation-test.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@
     Reference genome:
       class: File
       location: 'https://zenodo.org/record/4555735/files/NC_045512.2_reference.fasta?download=1'
+      filetype: fasta
```
Comment (mvdbeek) [inline workflows/sars-cov-2-variant-calling/sars-cov-2-consensus-from-variation/consensus-from-variation-test.yml:6]:
```suggestion
```

File: workflows/sars-cov-2-variant-calling/sars-cov-2-consensus-from-variation/consensus-from-variation-test.yml [hunk 2]
```diff
@@ -17,6 +18,7 @@
         - identifier: SRR11578257
           class: File
           path: test-data/final_snpeff_annotated_variants.vcf
+          filetype: vcf
```
Comment (mvdbeek) [inline workflows/sars-cov-2-variant-calling/sars-cov-2-consensus-from-variation/consensus-from-variation-test.yml:21]:
```suggestion
```

**Comment by wm75** [comment]
🎉 Mysterious, but at least we know now.

**Comment by mvdbeek** [comment]
Weird, seems to have been `fasta` that was the problem, since https://github.com/galaxyproject/iwc/pull/32/commits/6e554092cee24990649a71227af393455af153c9 worked!?

**Comment by wm75** [comment]
Oh, this is still open. Should be ready for merging?

**Pull request merged**